rpc/block.go

```diff
@@ -166,3 +167,25 @@ func (provider *Provider) BlockWithTxs(ctx context.Context, blockID BlockID) (in
 	}
 	return &result, nil
 }
+
+// Get block information with full transactions and receipts given the block id
+func (provider *Provider) BlockWithReceipts(ctx context.Context, blockID BlockID) (interface{}, *RPCError) {
+	var result json.RawMessage
+	if err := do(ctx, provider.c, "starknet_getBlockWithReceipts", &result, blockID); err != nil {
+		return nil, tryUnwrapToRPCErr(err, ErrBlockNotFound)
+	}
+	var block BlockWithReceipts
+	err := json.Unmarshal(result, &block)
+	if err == nil && block.BlockStatus != "" {
+		return &block, nil
+	}
+
+	// if result.Status == nil it's a pending block
```

Comment:
This is incorrect, please see [here](https://github.com/starkware-libs/starknet-specs/blob/7ae9e7bdcd2126bd30830b96b10932cf59d13ae3/api/starknet_api_openrpc.json#L3201). If the block is pending then the status should be `"PENDING"`.

Reply:
The naming could be clearer tbh, but I think the current implementation is correct.

Reply:
I've updated the logic to make this a bit clearer, but it's essentially the same

Reply:
Yeah looks like the spec is ambiguous